pxr/imaging/hdSt/materialXFilter.cpp

```diff
@@ -293,6 +480,57 @@ _AddDefaultMtlxTextureValues(std::map<TfToken, VtValue>* hdTextureParams)
     (*hdTextureParams) [_tokens->sourceColorSpace] = VtValue(HdStTokens->raw);
 }
 
+static void
+_AddDefaultMtlxTextureValues(
+    mx::NodeDefPtr const& nodeDef,
+    std::map<TfToken, VtValue>* hdTextureParams)
+{
+    _AddDefaultMtlxTextureValues(hdTextureParams);
+
+    if (nodeDef->getCategory() == mx::ShaderNode::IMAGE) {
```

Comment:
Isn't the category for nodeDef's 'nodeDef' should we be checking the category on the mtlx Node itself?

Reply:
Agreed. Checking on the node would indeed work. The other option is to use `nodeDef->getNodeString()` instead. Do you want me to update the PR?

Reply:
But, unit tests passing, as well as my test scenes rendering correctly, give a strong indication that this code block should be removed.

Reply:
I agree I think it should be removed. My next question was going to be how is it different from the other `_AddDefaultMtlxTextureValues()` above, which seems to be the reason the tests were ok. 
Am I understanding this function correctly in that the first `_AddDefaultMtlxTextureValues()` call handles the `<image>` nodes and the rest is intended to handle the `<tiledimage>` nodes as well as any other custom texture based nodes?

Reply:
We are looking for default values and indeed those default are correct for `mx::ShaderNode::IMAGE`. One more reason to remove the `if (nodeDef->getCategory() == mx::ShaderNode::IMAGE)` block of code.

The first `_AddDefaultMtlxTextureValues()` can probably be merged into the second one since it is the only place where it is called.

Reply:
I just browsed the MaterialX library, and all image nodes used in NodeGraphs are `repeat`, which mean the defaults are sufficient for all cases I tested.

The only way to insure this code actually does something useful would be to create a custom `untiledimage` NodeDef that does not have `repeat` as the default on its embedded image node.

Reply:
Just to verify my understanding of this function. Since the first `_AddDefaultMtlxTextureValues()` correctly gets the default values for both of the image based stdlib nodes, is the rest of the function (after that if block we agree should be removed) handling the case that we have a custom texture node that uses different default values for the wrap mode or filterType than the stdlib nodes?